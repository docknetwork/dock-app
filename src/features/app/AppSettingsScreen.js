import React from 'react';
import {
  Header,
  Content,
  ScreenContainer,
  Typography,
  Box,
  NBox,
  BackButton,
  OptionList,
  DownloadIcon,
  TrashIcon,
} from '../../design-system';
import {useDispatch} from 'react-redux';
import {AppConstants} from './constants';
import {walletOperations} from '../create-wallet/wallet-slice';

const constants = AppConstants.settings;

export function AppSettingsScreen({onDeleteWallet, onBackupWallet}) {
  return (
    <ScreenContainer testID="AccountDetailsScreen">
      <Header>
        <Box
          marginLeft={1}
          marginRight={22}
          flexDirection="row"
          alignItems="center">
          <NBox width={'80px'}>
            <BackButton />
          </NBox>
          <NBox
            flex={1}
            width="100%"
            alignContent="center"
            alignItems="center"
            pl={15}>
            <Typography.ScreenTitle>
              {constants.locales.title}
            </Typography.ScreenTitle>
          </NBox>
          <NBox width="80px" alignItems="flex-end" />
        </Box>
      </Header>
      <Content>
        <OptionList
          mx={5}
          items={[
            {
              testID: constants.testID.backupWalletOption,
              title: constants.locales.backupWallet,
              icon: <DownloadIcon />,
              onPress: onBackupWallet,
            },
            {
              testID: constants.testID.deleteWalletOption,
              title: constants.locales.deleteWallet,
              icon: <TrashIcon />,
              onPress: onDeleteWallet,
            },
          ]}
        />
      </Content>
    </ScreenContainer>
  );
}

export function AppSettingsContainer() {
  const dispatch = useDispatch();

  return (
    <AppSettingsScreen
      onDeleteWallet={() => {
        return dispatch(walletOperations.deleteWallet());
      }}
      onBackupWallet={() => {
        alert('Available soon');
      }}
    />
  );
}
