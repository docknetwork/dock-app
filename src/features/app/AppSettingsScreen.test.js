import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {AppSettingsScreen} from './AppSettingsScreen';

const mockStore = configureMockStore();

describe('AppSettingsScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<AppSettingsScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
