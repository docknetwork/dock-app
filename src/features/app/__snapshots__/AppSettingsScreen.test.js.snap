// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSettingsScreen should render correctly 1`] = `
<StyledNativeComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "inlineStyle": InlineStyle {
        "rules": Array [
          "
  background-color: ",
          "#18181B",
          ";
  flex: 1;
",
        ],
      },
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "StyledNativeComponent",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  testID="AccountDetailsScreen"
>
  <Styled(View)>
    <Box
      alignItems="center"
      flexDirection="row"
      marginLeft={1}
      marginRight={22}
    >
      <Memo(ForwardRef(Box))
        width="80px"
      >
        <BackButton />
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        alignContent="center"
        alignItems="center"
        flex={1}
        pl={15}
        width="100%"
      >
        <Styled(Component)
          variant="h3"
        >
          Settings
        </Styled(Component)>
      </Memo(ForwardRef(Box))>
      <Memo(ForwardRef(Box))
        alignItems="flex-end"
        width="80px"
      />
    </Box>
  </Styled(View)>
  <Styled(ScrollView)>
    <OptionList
      items={
        Array [
          Object {
            "icon": <SvgDownload />,
            "onPress": undefined,
            "testID": "backupWallet",
            "title": "Backup wallet",
          },
          Object {
            "icon": <SvgTrash />,
            "onPress": undefined,
            "testID": "deleteWallet",
            "title": "Delete wallet",
          },
        ]
      }
      mx={5}
    />
  </Styled(ScrollView)>
</StyledNativeComponent>
`;
